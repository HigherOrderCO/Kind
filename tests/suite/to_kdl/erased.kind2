// We should open an issue for this?

Arity3 -(e: U60) -(f: U60) <g> <h> <i> (d: U60) : U60
Arity3 e f g h i d = d

ArityB -(e: U60) <g> (d: U60) -(f: U60) <h> -(h: U60) <i> : U60
ArityB e g d f h i j = d

Main : U60
Main =
  let a = (+ (+ (Arity3 1 2 3) (Arity3 1 2 3 4 5 6)) 5)
  let b = (+ (+ (ArityB 1 2 3 4) (ArityB 1 2 3 4 5 6 7)) 5)
  b