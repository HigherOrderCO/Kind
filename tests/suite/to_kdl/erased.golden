// Arity3 #Arity3 -(e: U60) -(f: U60) -(g: Type) -(h: Type) -(i: Type) (d: U60) : U60
fun (Arity3 d) {
  (Arity3 d) = d
}

// ArityB #ArityB -(e: U60) -(g: Type) (d: U60) -(f: U60) -(h: Type) -(h: U60) -(i: Type) : U60
fun (ArityB d) {
  (ArityB d) = d
}

run {
  let ~ = (+ (+ (Arity3 #3) (Arity3 #6)) #5); let b = (+ (+ (ArityB #2) (ArityB #3)) #5); b
}

